packages/components/src/components/radio-button/radio-button.ts

```diff
@@ -0,0 +1,236 @@
+import { css, html } from 'lit';
+import { customElement } from '../../../src/internal/register-custom-element';
+import { HasSlotController } from '../../internal/slot';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { property, query, state } from 'lit/decorators.js';
+import { watch } from '../../internal/watch';
+import componentStyles from '../../styles/component.styles';
+import cx from 'classix';
+import SolidElement from '../../internal/solid-element';
+
+/**
+ * @summary Radio buttons allow the user to select a single option from a group using a button-like control.
+ * @documentation https://solid.union-investment.com/[storybook-link]/radio-button
+ * @status stable
+ * @since 1.0
+ *
+ * @slot - The radio button's label.
+ * @slot icon - A presentational icon.
+ *
+ * @event sd-blur - Emitted when the button loses focus.
+ * @event sd-focus - Emitted when the button gains focus.
+ *
+ * @csspart base - The component's base wrapper.
+ * @csspart button - The internal `<button>` element.
+ * @csspart button--checked - The internal button element when the radio button is checked.
+ * @csspart icon - The container that wraps the icon.
+ * @csspart label - The container that wraps the radio button's label.
+ */
+@customElement('sd-radio-button')
+export default class SdRadioButton extends SolidElement {
+  private readonly hasSlotController = new HasSlotController(this, '[default]', 'icon');
+
+  @query('.button') input: HTMLInputElement;
+  @query('.hidden-input') hiddenInput: HTMLInputElement;
+
+  @state() protected hasFocus = false;
+
+  /**
+   * @internal The radio button's checked state. This is exposed as an "internal" attribute so we can reflect it, making
+   * it easier to style in button groups.
+   */
+  @property({ type: Boolean, reflect: true }) checked = false;
+
+  /** The radio's value. When selected, the radio group will receive this value. */
+  @property() value: string;
+
+  /** Disables the radio button. */
+  @property({ type: Boolean, reflect: true }) disabled = false;
+
+  /** The radio button's size. */
+  @property({ reflect: true }) size: 'sm' | 'md' | 'lg' = 'lg';
+
+  /** Shows or hides the label */
+  @property({ type: Boolean }) showLabel = false;
```

Comment:
I have concern regarding the inconsistency in naming; one property is labeled as 'showLabel,' while the other is named 'hideIcon.' Could we standardize this and use either 'show' for both or 'hide' for both?

Reply:
I see your point, however it is also good practice to have the boolean props set to false by default? This was the reason for different naming (the icon is there by default and the label is not), but if it remains an issue maybe there's another way to approach it..

Reply:
@azraefendic your right (IMO). As defaults have to be false in a boolean attribute, this would be correct. 

in the radio-group we have:
```
  /**
   * The radio group's label. Required for proper accessibility. If you need to display HTML, use the `label` slot
   * instead.
   */
  @property() label = '';
```

Could we do something similar here as well?

Reply:
1. Yah, or setting a `label` property and the slot overrides it. I think that would be best DX to have it similar. And if there's nothing, it should be hidden by itself.

2. Yes, but that's the same for every button which only shows an icon.